src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTray.jsx

```diff
@@ -0,0 +1,105 @@
+import { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { useSelector } from 'react-redux';
+import { Button } from '@edx/paragon';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+
+import { useModel } from '../../../../../generic/model-store';
+import { LOADING } from '../../../../../course-home/data/slice';
+import Section from '../../../../../course-home/outline-tab/Section';
+import PageLoading from '../../../../../generic/PageLoading';
+import SidebarBase from '../../common/SidebarBase';
+import SidebarContext from '../../SidebarContext';
+import CourseOutlineTrigger, { ID } from './CourseOutlineTrigger';
+import messages from './messages';
+
+const CourseOutlineTray = ({ intl }) => {
+  const [expandAll, setExpandAll] = useState(false);
+
+  const {
+    courseStatus,
+  } = useSelector(state => state.courseHome);
+
+  const activeSequenceId = useSelector(state => state.courseware.sequenceId);
+
+  const {
+    courseId,
+    shouldDisplayFullScreen,
+  } = useContext(SidebarContext);
+
+  const {
+    courseBlocks,
+  } = useModel('outline', courseId);
+
+  const {
+    verifiedMode,
+  } = useModel('courseHomeMeta', courseId);
+
+  const {
+    sectionId: activeSectionId,
+  } = useModel('sequences', activeSequenceId);
+
+  const rootCourseId = courseBlocks?.courses && Object.keys(courseBlocks?.courses)[0];
+
+  if (courseStatus === LOADING) {
+    return (
+      <SidebarBase
+        title={intl.formatMessage(messages.courseOutlineTitle)}
+        ariaLabel={intl.formatMessage(messages.courseOutlineTray)}
+        sidebarId={ID}
+        className={classNames({
+          'h-100': !verifiedMode && !shouldDisplayFullScreen,
+          'mr-4': !shouldDisplayFullScreen,
+        })}
+      >
+        <PageLoading
+          srMessage={intl.formatMessage(messages.loading)}
+        />
+      </SidebarBase>
+    );
+  }
+
+  return (
+    <SidebarBase
+      title={intl.formatMessage(messages.courseOutlineTitle)}
+      ariaLabel={intl.formatMessage(messages.courseOutlineTray)}
+      sidebarId={ID}
+      className={classNames({
+        'h-100': !verifiedMode && !shouldDisplayFullScreen,
+        'mr-4': !shouldDisplayFullScreen,
+      })}
+    >
+      {rootCourseId && (
+        <>
+          <div className="row mb-3 justify-content-center">
+            <div className="col-12">
+              <Button variant="outline-primary" block onClick={() => { setExpandAll(!expandAll); }}>
```

Comment:
```suggestion
              <Button variant="outline-primary" block onClick={() => setExpandAll(!expandAll)}>
```

Reply:
This functional will be removed in the next PR: https://github.com/raccoongang/frontend-app-learning/pull/17